### PR TITLE
add SIDER

### PR DESCRIPTION
**Pull request by AdrianM0**
One more dataset from moleculeNet. 🥳

### CODE REVIEW

File: data/tabular/SIDER/transform.py [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+            "description": f"{description}",
+            "type": "boolean",
+            "names": [{"noun": f"{description}".lower()}],
+        }
+        for col_id, description in zip(column_ids[1:], descriptions[1:])
+    ]
+
+    templates = [
+        "The {#molecule|compound|chemical|molecular species|chemical compound!} with the {SMILES__description}"  # noqa: E501
+        + "{#representation of |!}{SMILES#} is {"
```
Comment (kjappelbaum) [inline data/tabular/SIDER/transform.py:38]:
```suggestion
        + "{#representation of |!} {SMILES#} is {"

```

File: data/tabular/SIDER/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,220 @@
...
+      type: boolean
+      names:
+          - noun: hepatobiliary disorders
+    - id: metabolism_and_nutrition_disorders
+      description: metabolism and nutrition disorders
+      type: boolean
+      names:
+          - noun: metabolism and nutrition disorders
+    - id: product_issues
+      description: product issues
```
Comment (kjappelbaum) [inline data/tabular/SIDER/meta.yaml:20]: what does this mean? what is an example of a product issue?

File: data/tabular/SIDER/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,220 @@
...
+      description: musculoskeletal and connective tissue disorders
+      type: boolean
+      names:
+          - noun: musculoskeletal and connective tissue disorders
+    - id: gastrointestinal_disorders
+      description: gastrointestinal disorders
+      type: boolean
+      names:
+          - noun: gastrointestinal disorders
+    - id: social_circumstances
```
[the remark this answers is not in the document]
Comment (kjappelbaum) [reply data/tabular/SIDER/meta.yaml:44]: or is this some kind of disorder?

[the remark this answers is not in the document]
Comment (AdrianM0) [reply data/tabular/SIDER/meta.yaml:44]: @kjappelbaum  I can also remove these two?

[the remark this answers is not in the document]
Comment (kjappelbaum) [reply data/tabular/SIDER/meta.yaml:44]: also an option if you cannot figure out what those are about

File: data/tabular/SIDER/meta.yaml (hunk 2 above)
Comment (kjappelbaum) [reply data/tabular/SIDER/meta.yaml:20]: @AdrianM0 do you know what product issues is supposed to mean? Does this noun fit the column name?

File: data/tabular/SIDER/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,206 @@
...
+    - id: investigations
+      description: eye disorders
+      type: boolean
+      names:
+          - noun: eye disorders
+    - id: musculoskeletal_and_connective_tissue_disorders
+      description: investigations
+      type: boolean
+      names:
+          - noun: investigations
```
Comment (kjappelbaum) [inline data/tabular/SIDER/meta.yaml:33]: isn't there a mismatch between your column names and the `names`? 
They seem one off. 

Generally, it is probably better if we can manage to get multiple names to increase the diversity of the training data

Comment (AdrianM0) [reply data/tabular/SIDER/meta.yaml]: fixed this now, added one noun to each

File: data/tabular/SIDER/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,231 @@
...
+      type: boolean
+      names:
+          - noun: eye disorders
+          - noun: ophthalmic disorders
+    - id: investigations
+      description: investigations
+      type: boolean
+      names:
+          - noun: investigations
+          - noun: medical investigations
```
Comment (kjappelbaum) [inline data/tabular/SIDER/meta.yaml:32]: do we want to sample this one/do you know what this one means?

File: data/tabular/SIDER/meta.yaml [hunk 6]
```diff
@@ -0,0 +1,231 @@
...
+      type: boolean
+      names:
+          - noun: nervous system disorders
+          - noun: disorders of the nervous system
+    - id: injury_poisoning_and_procedural_complications
+      description: injury, poisoning and procedural complications
+      type: boolean
+      names:
+          - noun: injury, poisoning and procedural complications
+          - noun: injuries, poisonings, and complications from procedures
```
Comment (kjappelbaum) [inline data/tabular/SIDER/meta.yaml:158]: "complications from procedures" -> this does not sound like a drug side effect to me ...

**Comment by kjappelbaum** [comment]
Thanks for all the updates, and sorry for being so picky!

File: data/tabular/SIDER/meta.yaml (hunk 6 above)
Comment (AdrianM0) [reply data/tabular/SIDER/meta.yaml:158]: refers to post-surgery complications likely

**Comment by AdrianM0** [comment]
@kjappelbaum this can be closed I think

**Comment by kjappelbaum** [comment]
With closed you mean merged?

File: data/tabular/SIDER/meta.yaml [hunk 7]
```diff
@@ -0,0 +1,223 @@
...
+      type: boolean
+      names:
+          - noun: nervous system disorders
+          - noun: disorders of the nervous system
+    - id: injury_poisoning_and_procedural_complications
+      description: injury, poisoning and procedural complications
+      type: boolean
+      names:
+          - noun: injury, poisoning and procedural complications
+          - noun: injuries and poisonings
```
Comment (kjappelbaum) [inline data/tabular/SIDER/meta.yaml:152]: i'm still not sure about this. You put in SMILES and then predict something about "procedural complications" which sounds more like surgery

Comment (kjappelbaum) [reply data/tabular/SIDER/meta.yaml]: @AdrianM0, what are your thoughts on this one?

**Comment by kjappelbaum** [comment]
can you, as discussed, remove the columns we are uncertain about? (https://discord.com/channels/@me/1138533229381550181/1168595210876485673)

**Review by kjappelbaum — APPROVED** [review]
thanks for the patience throughout all those updates!

**Pull request merged**